src/inmanta/deploy/state.py

```diff
@@ -177,6 +177,11 @@ class ModelState:
     version: int
     resources: dict[ResourceIdStr, ResourceDetails] = dataclasses.field(default_factory=dict)
     requires: RequiresProvidesMapping = dataclasses.field(default_factory=RequiresProvidesMapping)
+    """
+    Resources that have a new desired state (might be simply a change of its dependencies), which are still being processed by
+    the resource scheduler. This is a short-lived transient state, used for internal concurrency control. Kept separate from
+    ResourceStatus so that it lives outside of the scheduler lock's scope.
+    """
     resource_state: dict[ResourceIdStr, ResourceState] = dataclasses.field(default_factory=dict)
     # resources with a known or assumed difference between intent and actual state
     dirty: set[ResourceIdStr] = dataclasses.field(default_factory=set)
```

Comment:
I think this is describing the `dirty` set.
```suggestion
    resource_state: dict[ResourceIdStr, ResourceState] = dataclasses.field(default_factory=dict)
    # resources with a known or assumed difference between intent and actual state
    # (might be simply a change of its dependencies), which are still being processed by
    # the resource scheduler. This is a short-lived transient state, used for internal concurrency control. Kept separate from
    # ResourceStatus so that it lives outside of the scheduler lock's scope.
    dirty: set[ResourceIdStr] = dataclasses.field(default_factory=set)
```